examples/jaeger/main.cc

```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "opentelemetry/exporters/jaeger/jaeger_exporter_factory.h"
+#include "opentelemetry/sdk/trace/processor.h"
```

Comment:
We need to add "sdk/trace/exporter.h" too here ?

Reply:
I guess, I did not catch yet due to "order of refactoring header files" :).  Currently, I did it for only API and SDK folders but not for exporters folder. Probably, in that time, I will just remove **"sdk/trace/exporter.h"** file from **opentelemetry/exporters/jaeger/jaeger_exporter_factory.h** file and it will cause an compilation error in this file due to complete type problem.

Reply:
@lalitb @cngzhnp 

The scope of this cleanup is mostly the content of sdk, with minor adjustments as needed to other parts (exporter, examples) when needed.

Lalit comments are on the examples area which will be cleaned up later, sure, but the comments are still valid.

@cngzhnp Please implement comments as indicated, so this review can complete, and the patch be merged.

This will avoid the extra burden, for everyone, of keeping these comments aside, remember them, and either verify the fix later or raise comments again when the content of examples is fixed as well.

4 lines to fix in a patch touching 172 files is very minor, so looking forward to a quick resolution on this, so that @lalitb can approve the overall change, allowing the merge to main soon.

Thanks both the the massive cleanup and review.